### ISSUE TITLE
Check contents of /etc/resolv.conf

### ISSUE THREAD

**Issue by yrro** [freeipa/freeipa-healthcheck#209]
I've got a machine where `resolv.conf` was changed from:

```
# auto-generated by IPA installer
search ipa.example.com
nameserver 127.0.0.1
nameserver ::1
```

to:

```
; generated by /usr/sbin/dhclient-script
nameserver 203.0.113.1
nameserver 203.0.113.2
nameserver 203.0.113.3
```

Maybe `ipa-healthcheck` could complain if `::1` and `127.0.0.1` are not the only configured nameservers, and if the search list doesn't include the IPA domain?

**Comment by rcritten** [comment]
That's a great suggestion, thanks.

We'd need to be careful to only apply this when the DNS service is configured and to not assume which addresses are there.

The installer configures 127.0.0.1 if there are any IPv4 addresses and ::1 if any IPv6 in https://github.com/freeipa/freeipa/blob/master/ipaserver/install/bindinstance.py#L1127

**Comment by yrro** [comment]
Since (I think) I filed this issue, FreeIPA has been enhanced to tell NetworkManager to configure the system's DNS to use 127.0.0.1/::1 as its resolver. So probably the original reason I filed this issue is no longer reproducible.

Nonetheless it's probably still useful to have a health check for proper DNS configuration. If you want to check whether resolved 'owns `resolv.conf` then the best way to find out is to ask it via D-Bus (check the manager's `ResolvConfMode` property which can be set to `uplink`, `stub`, `static` if managed, `missing` if `resolv.conf` has been deleted, and `foreign` if `resolv.conf` is not managed by resolved).

**Comment by rcritten** [comment]
Yes, D-Bus is a much nicer way to inquire on status, thanks for that!

**Comment by rcritten** [comment]
I may add this under meta as an informational value.

Simple code for getting this property:
```
import dbus

bus = dbus.SystemBus()
obj = bus.get_object('org.freedesktop.resolve1',
                     '/org/freedesktop/resolve1')
prop_if = dbus.Interface(obj, 'org.freedesktop.DBus.Properties')
mode = prop_if.Get('org.freedesktop.resolve1.Manager', 'ResolvConfMode')
print(mode)
```
